src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -190,6 +209,62 @@ trait AttributeComponent {
       filter(_.id inSetBind attributeRecords.map(_.id)).delete
     }
 
+    object AlterAttributesUsingTempTableQueries extends RawSqlQuery {
+      val driver: JdbcDriver = AttributeComponent.this.driver
+      /*!CAUTION! - do not drop the ownerIds clause at the end or it will delete every attribute in the table!
+
+        deleteMasterFromAction: deletes any row in *_ATTRIBUTE that doesn't
+          exist in *_ATTRIBUTE_TEMP but exists in *_ATTRIBUTE (bound by ownerIds)
+        updateInMasterAction: updates any row in *_ATTRIBUTE that also
+          exists in *_ATTRIBUTE_TEMP (bound by ownerIds)
+        insertIntoMasterAction: inserts any row into *_ATTRIBUTE that doesn't
+          exist in *_ATTRIBUTE but exists in *_ATTRIBUTE_TEMP (bound by ownerIds)
+
+        These left joins give us the rows related to the attributes that are in the temp table.
+        We filter on ta.owner_id == null if we want to see that the row does NOT exist in the temp table
+        We filter on ea.owner_id == null if we want to see that the row does NOT exist in the real table
+        We filter on ta.owner_id != null if we want to see that the row exists in both of the tables
+       */
+      def deleteFromMasterAction(ownerIds: String) =
```

Comment:
these should take in a collection of owner ids, not a string with preformed sql, then the trick is figuring out the SetParameter implicit